tests: erratic failure of TestReconfigure

on travis-ci. log:

```
--- FAIL: TestReconfigure (0.64s)
	logger.go:16: INFO: nsqlookupd v1.0.0-compat (built w/go1.8.5)
	logger.go:16: INFO: nsqlookupd v1.0.0-compat (built w/go1.8.5)
	logger.go:16: INFO: nsqlookupd v1.0.0-compat (built w/go1.8.5)
	logger.go:16: INFO: HTTP: listening on 127.0.0.1:35355
	logger.go:16: INFO: TCP: listening on 127.0.0.1:39363
	logger.go:16: INFO: HTTP: listening on 127.0.0.1:38923
	logger.go:16: INFO: nsqd v1.0.0-compat (built w/go1.8.5)
	logger.go:16: INFO: TCP: listening on 127.0.0.1:44689
	logger.go:16: INFO: HTTP: listening on 127.0.0.1:34465
	logger.go:16: INFO: ID: 789
	logger.go:16: INFO: TCP: listening on 127.0.0.1:46145
	logger.go:16: INFO: TCP: listening on 127.0.0.1:42017
	logger.go:16: INFO: HTTP: listening on 127.0.0.1:37157
	logger.go:16: INFO: LOOKUP(127.0.0.1:39363): adding peer
	logger.go:16: INFO: LOOKUP connecting to 127.0.0.1:39363
	logger.go:16: INFO: TCP: new client(127.0.0.1:58032)
	logger.go:16: INFO: CLIENT(127.0.0.1:58032): desired protocol magic '  V1'
	logger.go:16: INFO: CLIENT(127.0.0.1:58032): IDENTIFY Address:travis-job-nsqio-nsq-297641244.travisci.net TCP:42017 HTTP:37157 Version:1.0.0-compat
	logger.go:16: INFO: DB: client(127.0.0.1:58032) REGISTER category:client key: subkey:
	logger.go:16: INFO: LOOKUPD(127.0.0.1:39363): peer info {TCPPort:39363 HTTPPort:35355 Version:1.0.0-compat BroadcastAddress:travis-job-nsqio-nsq-297641244.travisci.net}
	logger.go:16: INFO: LOOKUP(127.0.0.1:39363): removing peer
	logger.go:16: INFO: LOOKUP(127.0.0.1:44689): adding peer
	logger.go:16: INFO: LOOKUP connecting to 127.0.0.1:44689
	logger.go:16: INFO: CLIENT(127.0.0.1:58032): closing
	logger.go:16: INFO: DB: client(127.0.0.1:58032) UNREGISTER category:client key: subkey:
	logger.go:16: ERROR: client(127.0.0.1:58032) - EOF
	logger.go:16: INFO: TCP: new client(127.0.0.1:48424)
	logger.go:16: INFO: CLIENT(127.0.0.1:48424): desired protocol magic '  V1'
	logger.go:16: INFO: CLIENT(127.0.0.1:48424): IDENTIFY Address:travis-job-nsqio-nsq-297641244.travisci.net TCP:42017 HTTP:37157 Version:1.0.0-compat
	logger.go:16: INFO: DB: client(127.0.0.1:48424) REGISTER category:client key: subkey:
	logger.go:16: INFO: LOOKUPD(127.0.0.1:44689): peer info {TCPPort:44689 HTTPPort:38923 Version:1.0.0-compat BroadcastAddress:travis-job-nsqio-nsq-297641244.travisci.net}
	logger.go:16: INFO: LOOKUP(127.0.0.1:46145): adding peer
	logger.go:16: INFO: LOOKUP connecting to 127.0.0.1:46145
	assertions.go:14: nsqd_test.go:396:

			   2 (expected)

			!= 1 (actual)

	logger.go:16: INFO: NSQ: persisting topic/channel metadata to /tmp/nsq-test-329291789/nsqd.dat
	logger.go:16: INFO: TCP: closing 127.0.0.1:42017
	logger.go:16: INFO: HTTP: closing 127.0.0.1:37157
```

--- a/nsqd/nsqd.go
+++ b/nsqd/nsqd.go
@@ -92,6 +92,8 @@ func New(opts *Options) *NSQD {
 	httpcli := http_api.NewClient(nil, opts.HTTPClientConnectTimeout, opts.HTTPClientRequestTimeout)
 	n.ci = clusterinfo.New(n.logf, httpcli)
 
+	n.lookupPeers.Store([]*lookupPeer{})
+
 	n.swapOpts(opts)
 	n.errValue.Store(errStore{})
 

--- a/nsqd/nsqd_test.go
+++ b/nsqd/nsqd_test.go
@@ -368,32 +368,43 @@ func TestReconfigure(t *testing.T) {
 	defer os.RemoveAll(opts.DataPath)
 	defer nsqd.Exit()
 
-	time.Sleep(250 * time.Millisecond)
-
-	newOpts := *opts
+	newOpts := NewOptions()
+	newOpts.Logger = opts.Logger
 	newOpts.NSQLookupdTCPAddresses = []string{lookupd1.RealTCPAddr().String()}
-	nsqd.swapOpts(&newOpts)
+	nsqd.swapOpts(newOpts)
 	nsqd.triggerOptsNotification()
 	test.Equal(t, 1, len(nsqd.getOpts().NSQLookupdTCPAddresses))
 
-	time.Sleep(350 * time.Millisecond)
-
-	numLookupPeers := len(nsqd.lookupPeers.Load().([]*lookupPeer))
+	var numLookupPeers int
+	for i := 0; i < 100; i++ {
+		numLookupPeers = len(nsqd.lookupPeers.Load().([]*lookupPeer))
+		if numLookupPeers == 1 {
+			break
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
 	test.Equal(t, 1, numLookupPeers)
 
-	newOpts = *opts
+	newOpts = NewOptions()
+	newOpts.Logger = opts.Logger
 	newOpts.NSQLookupdTCPAddresses = []string{lookupd2.RealTCPAddr().String(), lookupd3.RealTCPAddr().String()}
-	nsqd.swapOpts(&newOpts)
+	nsqd.swapOpts(newOpts)
 	nsqd.triggerOptsNotification()
 	test.Equal(t, 2, len(nsqd.getOpts().NSQLookupdTCPAddresses))
 
-	time.Sleep(350 * time.Millisecond)
+	for i := 0; i < 100; i++ {
+		numLookupPeers = len(nsqd.lookupPeers.Load().([]*lookupPeer))
+		if numLookupPeers == 2 {
+			break
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
+	test.Equal(t, 2, numLookupPeers)
 
 	var lookupPeers []string
 	for _, lp := range nsqd.lookupPeers.Load().([]*lookupPeer) {
 		lookupPeers = append(lookupPeers, lp.addr)
 	}
-	test.Equal(t, 2, len(lookupPeers))
 	test.Equal(t, newOpts.NSQLookupdTCPAddresses, lookupPeers)
 }
 

